apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.html.ts

```diff
@@ -0,0 +1,674 @@
+  defaultMin: z.number().optional(),
+  defaultMax: z.number().optional(),
+});
+type GroupConfigInfo = z.infer<typeof GroupConfigInfoSchema>;
+
+export const GroupUsersRowSchema = z.object({
+  group_id: IdSchema,
+  name: z.string(),
+  size: z.number(),
+  users: z.array(UserSchema.pick({ user_id: true, uid: true })),
+});
+type GroupUsersRow = z.infer<typeof GroupUsersRowSchema>;
+
+export function InstructorAssessmentGroups({
+  resLocals,
+  groupsCsvFilename,
+  groupConfigInfo,
+  groups,
+  notAssigned,
+}: {
+  resLocals: Record<string, any>;
+  groupsCsvFilename?: string;
+  groupConfigInfo?: GroupConfigInfo;
+  groups?: GroupUsersRow[];
+  notAssigned?: string[];
+}) {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../partials/head'); %>", resLocals)}
+        <link
+          href="${nodeModulesAssetPath('tablesorter/dist/css/theme.bootstrap.min.css')}"
+          rel="stylesheet"
+        />
+        <script src="${nodeModulesAssetPath(
+            'tablesorter/dist/js/jquery.tablesorter.min.js',
+          )}"></script>
+        <script src="${nodeModulesAssetPath(
+            'tablesorter/dist/js/jquery.tablesorter.widgets.min.js',
+          )}"></script>
+      </head>
+
+      <body>
+        <script>
+          $(function () {
+            $('[data-toggle="popover"]').popover({ sanitize: false });
+
+            // Prevent the dropdown menu from closing when the popover is opened.
+            $('.js-group-action[data-toggle="popover"]').on('click', (e) => {
+              e.stopPropagation();
+            });
+
+            $('.js-group-action-dropdown').on('hide.bs.dropdown', (e) => {
+              // If the click is inside a popover, don't hide the dropdown.
+              if (e.clickEvent.target.closest('.popover')) {
+                e.preventDefault();
+                e.stopPropagation();
+                return;
+              }
+
+              // Hide all popovers when the dropdown menu is closed.
+              $('.js-group-action[data-toggle="popover"]').popover('hide');
+            });
+          });
+
+          // make the file inputs display the file name
+          $(document).on('change', '.custom-file-input', function () {
+            this.fileName = $(this).val().replace(/\\\\/g, '/').replace(/.*\\//, '');
+            $(this).parent('.custom-file').find('.custom-file-label').text(this.fileName);
+          });
+        </script>
+        ${renderEjs(__filename, "<%- include('../partials/navbar'); %>", resLocals)}
+        <main id="content" class="container-fluid">
+          ${renderEjs(
+            __filename,
+            "<%- include('../partials/assessmentSyncErrorsAndWarnings'); %>",
+            resLocals,
+          )}
+          ${typeof resLocals.errormsg !== 'undefined' && resLocals.errormsg.length > 0
+            ? html` <div class="alert alert-danger" role="alert">${resLocals.errormsg}</div> `
+            : ''}
```

Comment:
Please double-check, but I believe this can be removed. This was from a previous version of this page that existed before flash messages were being used.

Reply:
This seems safe to be removed. Removed in [39b5e80](https://github.com/PrairieLearn/PrairieLearn/pull/9550/commits/39b5e80a1ce76beee898fed6daf650f981c27704).